reo/src/reopt.jl

```diff
@@ -644,16 +644,48 @@ function add_tou_demand_charge_constraints(m, p)
 		sum( m[:dvGridPurchase][u, ts] for u in p.PricingTier )
 	)
 	
-	##Constraint (12e): Peak demand used in percent lookback calculation 
-	@constraint(m, [m in p.DemandLookbackMonths],
-		m[:dvPeakDemandELookback] >= sum(m[:dvPeakDemandEMonth][m, n] for n in p.DemandMonthsBin)
-	)
-	
-	##Constraint (12f): Ratchet peak demand charge is bounded below by lookback
-	@constraint(m, [r in p.DemandLookbackMonths],
-		sum( m[:dvPeakDemandEMonth][r,e] for e in p.DemandBin ) >= 
-		p.DemandLookbackPercent * m[:dvPeakDemandELookback] 
-	)
+	if p.DemandLookbackRange != 0  # then the dvPeakDemandELookback varies by month
+		
+		##Constraint (12e): dvPeakDemandELookback is the highest peak demand in DemandLookbackMonths
+		for mth in p.Month
+			if p.Month > p.DemandLookbackRange
+				@constraint(m, [lm in 1:p.DemandLookbackRange],
+					m[:dvPeakDemandELookback][mth] >= 
+					sum(m[:dvPeakDemandEMonth][mth - lm, n] for n in p.DemandMonthsBin)
+				)
```

Comment:
Instead of setting the dvPeakDemandELookback for a given month based on the relevant lookback months' dvPeakDemandEMonth, might it make sense to set it based on the lookback months' peak dvGridPurchase (might require a new variable, e.g. dvPeakDemandEMonthActual)? My concern is that using dvPeakDemandEMonth, peaks will be propogated through the whole year.

Reply:
For example: Let's say we have a 3-month rolling lookback of 100%. In August, dvPeakDemandELookback (August) and therefore dvPeakDemandEMonth(August) would be set based on May-July and August. So if the June peak is higher than the August peak, dvPeakDemandEMonth(August) would be set based on the June peak (which is what we want). However, then let's look at October.  dvPeakDemandELookback (October) and therefore dvPeakDemandEMonth(October) should be set based on July-September and October. However, since dvPeakDemandEMonth(August) is being set by the June peak, if dvPeakDemandELookback (October) is set based on dvPeakDemandEMonth then June's peak would be propagated through to October, even though June is more than 3 months before October.

Reply:
Yes that makes sense. I just committed a change using `dvGridPurchase`, thank you!